datatypes/src/util/arrow.rs

```diff
@@ -33,6 +33,15 @@ pub trait ArrowTyped {
     /// Return the specific arrow data type
     fn arrow_data_type() -> DataType;
 
+    fn arrow_list_data_type() -> DataType {
+        let nullable = true; // TODO: should actually be false, but arrow's builders set it to `true` currently
```

Comment:
why does this always have to be false? we do allow nulls for some types

Reply:
This trait is currently implemented for types like Coordinate, TimeInterval, MultiPoint where we don't allow nulls.
But yeah, in the future this could change, but for now, it has to stay nullable because of builder limitations.